tests/unit/s2n_ktls_cmsg_test.c

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <sys/socket.h>
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_ktls.h"
+#include "utils/s2n_random.h"
+
+S2N_RESULT s2n_ktls_send_msg_impl(int sock, struct msghdr *msg,
+        struct iovec *msg_iov, size_t count, s2n_blocked_status *blocked, ssize_t *send_len);
+S2N_RESULT s2n_ktls_recv_msg_impl(struct s2n_connection *conn, int sock, struct msghdr *msg,
+        struct iovec *msg_iov, s2n_blocked_status *blocked, ssize_t *bytes_read);
+S2N_RESULT s2n_ktls_set_ancillary_data(struct msghdr *msg, uint8_t record_type);
+S2N_RESULT s2n_ktls_parse_ancillary_data(struct msghdr *msg, uint8_t *record_type);
```

Comment:
You're doing the model where you test all the component private methods separately again instead of testing the methods of interest :/ And if you can't actually call s2n_ktls_send_msg / s2n_ktls_recv_msg, doesn't that also prevent you from properly testing any method that calls them? What is your plan for the next layer of tests?

Maybe you need to reuse the refactor from https://github.com/aws/s2n-tls/pull/4071/files#r1268745257 for sendmsg and recvmsg. And at that point you've probably got too many "mocks" to reasonably argue they aren't just for testing, so I'd probably switch to the global version instead of the config version.